### PR TITLE
JavaListener memory usage reduction

### PR DESCRIPTION
**Pull request by tnleeuw**

### CODE REVIEW

**Comment by tnleeuw** [comment]
I did not add unit-tests in this PR because I'm touching quite a lot of existing code all over the place that just needs a few changes in how API is called.

However the code has very poor unit test coverage so coverage in the PR is low. I'm not sure how much effort to put in for unit tests.

I've tested everything with IAF-Test until that worked.

**Comment by tnleeuw** [comment]
Now including a good amount of unit tests.

File: core/src/main/java/nl/nn/adapterframework/receivers/JavaListener.java [hunk 1]
```diff
@@ -113,19 +113,30 @@ public synchronized void close() throws ListenerException {
...
+			Message result = processRequest(correlationId, (M) rawMessage, new Message(rawMessage), context);
+			return result.asString();
+		} catch (IOException e) {
+			throw new ListenerException("cannot convert stream", e);
+		}
+	}
+
+	@Override
+	public Message processRequest(String correlationId, Message message, PipeLineSession session) throws ListenerException {
+		return processRequest(correlationId, (M) message.asObject(), message, session);
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/receivers/JavaListener.java:136]: Ik denk dat het verstandig is de response hier te registreren als closeOnClose op de nieuwe session.
```suggestion
Message response = processRequest(correlationId, (M) message.asObject(), message, session);
session.closeOnClose(response);
return  response;
```

File: core/src/main/resources/SpringApplicationContext.xml [hunk 2]
```diff
@@ -104,7 +104,7 @@
 
 	<bean
 		name="isolatedServiceCaller"
-		class="nl.nn.adapterframework.pipes.IsolatedServiceCaller"
+		class="nl.nn.adapterframework.senders.IsolatedServiceCaller"
```
Comment (nielsm5) [inline core/src/main/resources/SpringApplicationContext.xml:107]: Moet de springIbisDebuggerAdvice ook aangepast worden?

Comment (tnleeuw) [reply core/src/main/resources/SpringApplicationContext.xml:107]: Voor zover ik kan zien niet

**Review by nielsm5 — APPROVED** [review]

**Pull request merged**